src/language/language.rs

```diff
@@ -523,6 +563,7 @@ pub enum MyAnalysisData {
     Legacy(MyAnalysisDataLegacyData),
     AccessPattern(AccessPatternData),
     Shape(ShapeData),
+    Tuple(Vec<Box<MyAnalysisData>>),
```

Comment:
Actually, I know I suggested `Vec<Box<...>>`, but a `Vec<MyAnalysisData>` might work just fine -- can you try that?

Reply:
BTW, the whole point of this is that you can't have directly recursive data structures in Rust, as Rust will have no idea what the stack size of the data structure is at compile time. But a Vec is allocated on the heap, so I'm pretty sure its stack size is known at compile time. I could be wrong.

Reply:
ah, I thought the problem was with enums of different sizes but it turns out all instantiations of an enum have the same size